src/main/java/backend/model/sponsoring/Sponsoring.kt

```diff
@@ -37,15 +39,18 @@ class Sponsoring : BasicEntity, Billable {
     lateinit var limit: Money
         private set
 
-    @ManyToOne
-    var team: Team? = null
+    @ManyToMany
+    var teams: MutableSet<Team> = HashSet()
 
     @ManyToOne(cascade = [PERSIST])
     private var unregisteredSponsor: UnregisteredSponsor? = null
 
     @ManyToOne
     private var registeredSponsor: Sponsor? = null
 
+    @ManyToOne
+    var event: Event? = null
```

Comment:
This is new because the event was initially read from the single team's event field. However, we theoretically might have 0 teams and thus still need to store the event relation somewhere.